pkg/nsx/services/staticroute/staticroute.go

```diff
@@ -134,12 +135,17 @@ func (service *StaticRouteService) GetUID(staticroute *model.StaticRoutes) *stri
 
 }
 
-func (service *StaticRouteService) DeleteStaticRoute(namespace string, uid string) error {
-	vpc := service.VPCService.ListVPCInfo(namespace)
-	if len(vpc) == 0 {
+func (service *StaticRouteService) DeleteStaticRoute(uid string) error {
+	id := String(util.GenerateID(uid, "sr", "", ""))
```

Comment:
I see we already pass UID to DeleteStaticRoute
https://github.com/vmware-tanzu/nsx-operator/pull/632/files#diff-98a294a67dfe8f3714cd9405624de61b68e401d40cc3c4c26e4c172cc4da2287R116

Why we re generate uid here by calling util.GenerateID, I think they are the same.
So why not just using uid passed in?